hangman.py

```diff
@@ -0,0 +1,427 @@
+            print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+
+        if is_word_guessed(secret_word, letters_guessed):
+            total_score = guesses_left * score(secret_word, letters_guessed)
+            print("Congratulations, you won! Your total score for this game is:", total_score)
+            break
+
+
+
+# When you've completed your hangman function, scroll down to the bottom
+# of the file and uncomment the first two lines to test
+#(hint: you might want to pick your own
+# secret_word while you're doing your own testing)
+
+
+# -----------------------------------
+
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    my_word_list = []
+    other_word_list = []
+
+    for x in my_word:
+        if x == ' ':
+            continue
+        my_word_list.append(x)
+
+    # Потрібно прописати випадок для 'a_ ple' 'apple'
+    for x in other_word:
+        other_word_list.append(x)
+
+    if len(my_word_list) != len(other_word_list):
+        return False
+
+    for x in range(0, len(my_word_list)):
+        if my_word_list[x] == '_' or my_word_list[x] == other_word_list[x]:
+            continue
+        else:
+            return False
+
+    my_word_list_without_gaps = []
+    for x in my_word_list:
+        if x == '_':
+            continue
+        else:
+            my_word_list_without_gaps.append(x)
+
+    my_word_list = my_word_list_without_gaps
+
+    for x in range(0, len(my_word_list)):
+        z = my_word_list.count(my_word_list[x])
+        y = other_word_list.count(my_word_list[x])
+        if z == y:
+            continue
+        else:
+            return False
+
+    return True
+
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    hint_list = []
+    for other_word in wordlist:
+        if match_with_gaps(my_word, other_word):
+            hint_list.append(other_word)
+    if hint_list == []:
+        print("No matches found")
+    else:
+        print("Possible word matches are:", ' '.join(hint_list))
+
+
+def hangman_with_hints(secret_word):
```

Comment:
This function is mostly the same as `hangman`. Try to avoid code duplication - combine these functions

Reply:
done